### PR TITLE
remove docs for old dplyr compat

### LINKED ISSUE #388: Remove help page on dplyr compatibility
[`R/compat-dplyr.R`](https://rsample.tidymodels.org/dev/reference/rsample-dplyr.html) talks about version-specific behaviour for dplyr versions older than `v1.0.0` but by now we require dplyr v1.0.0 so we can remove that.

### PR DESCRIPTION
**Pull request by hfrick**
closes  #388

I think this is a leftover from  #306 where the compat code for old (< v1.0.0) dplyr versions was removed.

### CODE REVIEW

**Comment by DavisVaughan** [comment]
i dont think we should just nuke these docs. i think they provide valuable rationale for why the operations return tibbles vs rset objects in the current version of rsample/dplyr. they probably just need a refresh where we remove sentences where we talk about old dplyr/rsample, and remove columns from the tables that talk about them

**Comment by hfrick** [comment]
Okay, thanks! The info about what happens with different versions was what I was focussed on, but I agree, the info on what happens with current versions of rsample and dplyr is still good to know. I've brought back that content and updated it accordingly.

File: R/compat-dplyr.R [hunk 1]
```diff
@@ -1,106 +1,62 @@
 #' Compatibility with dplyr
 #'
 #' @description
-#' rsample should be fully compatible with dplyr 1.0.0.
+#' This page lays out the compatibility between rsample and dplyr. The `rset`
+#' objects from rsample are a sepeciifc subclass of tibbles, hence standard
+#' dyplyr operations like joins as well row or column modifications work.
```
Comment (DavisVaughan) [inline R/compat-dplyr.R:6]:
```suggestion
#' objects from rsample are a specific subclass of tibbles, hence standard
#' dplyr operations like joins as well row or column modifications work.
```

Maybe do one more full pass for typos?

**Review by DavisVaughan — APPROVED** [review]

File: R/compat-dplyr.R (hunk 1 above)
Comment (hfrick) [reply R/compat-dplyr.R]: thanks!

**Pull request merged**